lib/src/metta/runner/stdlib/atom.rs

```diff
@@ -548,7 +548,61 @@ mod tests {
         assert_eq!(run_program("!(eval (foldl-atom (1 2 3) 0 $a $b (eval (+ $a $b))))"), Ok(vec![vec![expr!({Number::Integer(6)})]]));
     }
 
+    #[test]
+    fn metta_add_reducts() {
+        assert_eq!(run_program(&format!("!(chain (eval (new-space)) $newspace (add-reducts k1))")), Ok(vec![vec![expr!("Error" ("add-reducts" "k1") "IncorrectNumberOfArguments")]]));
```

Comment:
Taking into account https://github.com/trueagi-io/hyperon-experimental/pull/907#issuecomment-2783637378 I am not sure if `chain/eval` is a good pattern here. On the one hand it is safe because `(new-space)` is a grounded call and thus `eval` evaluates it in one step. On the other hand it is not generally evaluates `eval`'s argument till the end thus it is not possible using it everywhere.

Reply:
So, your suggestion is to replace chain-eval with let here just in case?